src/auth/googleauth.ts

```diff
@@ -243,34 +244,11 @@ export class GoogleAuth {
    * @returns A promise that resolves with the boolean.
    * @api private
    */
-  async _checkIsGCE(isRetry = false): Promise<boolean> {
+  async _checkIsGCE(): Promise<boolean> {
     if (this.checkIsGCE !== undefined) {
       return this.checkIsGCE;
     }
-    if (!this.transporter) {
-      this.transporter = new DefaultTransporter();
-    }
-    try {
-      const res = await this.transporter.request(
-          {url: 'http://metadata.google.internal'});
-      this.checkIsGCE =
-          res && res.headers && res.headers['metadata-flavor'] === 'Google';
-    } catch (e) {
-      const isDNSError = (e as NodeJS.ErrnoException).code === 'ENOTFOUND';
-      const ae = e as AxiosError;
-      const is5xx = ae.response &&
-          (ae.response.status >= 500 && ae.response.status < 600);
-      if (is5xx) {
-        // Unexpected error occurred. Retry once.
-        if (!isRetry) {
-          return await this._checkIsGCE(true);
-        }
-        throw e;
-      } else if (!isDNSError) {
-        throw e;
-      }
-      this.checkIsGCE = false;
-    }
+    this.checkIsGCE = await gcpMetadata.isAvailable();
```

Comment:
nit: Can this method be simplified to:

```js
return this.checkIsGCE || await gcpMetadata.isAvailable();
```

Reply:
That would get rid of the caching.  I was able to simplify it a bit though :)

Reply:
Why would it get rid of caching? The right hand side isn't evaluated if the left hand side is truthy.

Reply:
This would return the value before stashing it in `this.checkIsGCE`, wouldn't it?

Reply:
You're right. Your version is better.

```js
this.checkIsGCE = this.checkIsGCE || await gcpMetadata.isAvailable();
return this.checkIsGCE;
```

Reply:
Fair point :) Will update in a follow up PR!

Reply:
Seriously, I think your version as committed is better.